fzf-tab.zsh

```diff
@@ -374,24 +374,31 @@ fzf-tab-complete() {
 zle -N fzf-tab-complete
 
 disable-fzf-tab() {
-    typeset -g _ZSH_FZF_TAB_DISABLED
-    emulate -L zsh
-    if (( $+_fzf_tab_orig_widget )); then
-        bindkey '^I' $_fzf_tab_orig_widget
-        unset _fzf_tab_orig_widget
-    fi
+    emulate -L zsh -o extended_glob
+    (( $+_fzf_tab_orig_widget )) || return 0
+
+    bindkey '^I' $_fzf_tab_orig_widget
     case $_fzf_tab_orig_list_grouped in
         0) zstyle ':completion:*' list-grouped false ;;
         1) zstyle ':completion:*' list-grouped true ;;
         2) zstyle -d ':completion:*' list-grouped ;;
     esac
-    unset _fzf_tab_orig_list_groupded
+    unset _fzf_tab_orig_widget _fzf_tab_orig_list_groupded
+
+    # Don't remove .fzf-tab-orig-$_fzf_tab_orig_widget as we won't be able to reliably
+    # create it if enable-fzf-tab is called again.
 }
 
 enable-fzf-tab() {
-    unset _ZSH_FZF_TAB_DISABLED
-    emulate -L zsh
-    typeset -g _fzf_tab_orig_widget="${$(bindkey '^I')##* }"
+    emulate -L zsh -o extended_glob
+    (( $+_fzf_tab_orig_widget )) && return
+
+    typeset -g _fzf_tab_orig_widget="${${$(bindkey '^I')##* }:-expand-or-complete}"
+    if (( ! $+widgets[.fzf-tab-orig-$_fzf_tab_orig_widget] )); then
+      # Copy the widget before it's wrapped by zsh-autosuggestions and zsh-syntax-highlighting.
+      # Prefix the name of the copy with '.' so that it doesn't get wrapped.
```

Comment:
This will also prevent `compinit` from editing the widget, which will make fzf-tab unusable if it initializes before call `compinit`.

Reply:
Good point!

How about now?

Reply:
LGTM